client/ladder/ladder-reducer.ts

```diff
@@ -15,14 +15,33 @@ export interface SearchResults extends RetrievedRankings {
   searchQuery: string
 }
 
+// NOTE(2Pac): The reason why we have two separate maps for rankings and search results is so we
+// we don't have to re-fetch rankings when user clears the search query. Also, we keep the rankings
+// for current season separately from previous seasons because we don't immediately know the current
+// season ID.
 export interface LadderState {
+  /** A map of matchmaking type -> rankings. This is meant to be used for the current season. */
   typeToRankings: Map<MatchmakingType, RetrievedRankings>
+  /**
+   * A map of matchmaking type -> search results. This is meant to be used for the current season.
+   */
   typeToSearchResults: Map<MatchmakingType, SearchResults>
+  /**
+   * A map of season ID -> matchmaking type -> rankings. This is meant to be used for past seasons.
+   */
+  seasonToTypeToRankings: Map<SeasonId, Map<MatchmakingType, RetrievedRankings>>
```

Comment:
I would again prefer you convert these keys to strings rather than maintain nested maps